.github/workflows/deployToDev.yaml

```diff
@@ -0,0 +1,48 @@
+name: deploy to dev
```

Comment:
Can we modify https://github.com/NethermindEth/argo/blob/juno_main/.github/workflows/dispatch_render_juno.yml to use workflow_call instead of adding YMLs with curl dispatch? This way, we could directly call it from `build-and-deploy.yml`.

Reply:
@wojciechos Can we close this and merge this?

another PR will be made for Argo repo refactor.

cc: @Manjeet-Nethermind

Reply:
@NetherBob I think wojciechos' comment is valid. 

It makes more sense to change in this PR and resolve the code conflict than create a new PR to handle above change.

Anyway, code changes should be done in two repos (juno and argo). 
1. Argo repo: Refactor repository_dispatch method with workflow_call in   https://github.com/NethermindEth/argo/blob/main/.github/workflows/dispatch_render_juno.yml. New PR is required in argo repo.
2. Replace API(https://api.github.com/repos/NethermindEth/argo/dispatches) call with workflows caller in this PR. The workflow caller point may locate in deployToDev.yaml/deployToStaging.yaml/deployToProduction.yaml or build-and-deploy.yml
_run: |
              curl -L \
              -X POST \
              -H "Accept: application/vnd.github+json" \
              -H "Authorization: token ${{ secrets.ACCESS_TOKEN }}" \
              -H "X-GitHub-Api-Version: 2022-11-28" \
              https://api.github.com/repos/NethermindEth/argo/dispatches \
              -d '{"event_type": "${{ env.EVENT_NAME }}", "client_payload":{"name": "${{ env.EVENT_NAME }}", "goerli_config": "${{ env.GOERLI }}", "integration_config": "${{ env.INTEGRATION }}", "mainnet_config": "${{ env.MAINNET }}", "tag": "${{ env.IMAGE_TAG }}"}}'_

Reply:
@PhilexWong first of all, there isn't code conflict on the repo or PR, that's why it is a refactor.
And so as not to have too many changes littered around in a single PR, It is best practice to push and merge changes small and traceable. This is to avoid any error that may occur and can be easily rolled back.

I'm already working on the one on Argo and will create a PR once that is done.

cc: @wojciechos